scripts/bump.py

```diff
@@ -0,0 +1,281 @@
+
+
+def bump_tox(dependencies: t.Dict[str, str]) -> None:
+    """Bump tox file."""
+    if not TOX_INI.exists():
+        return
+
+    _logger.info("Updating tox.ini")
+    updated = ""
+    content = TOX_INI.read_text(encoding="utf-8")
+    for line in content.split("\n"):
+        try:
+            spec = Dependency.from_string(line.lstrip().rstrip())
+            update = dependencies.get(spec.name)
+            if update is None:
+                updated += line + "\n"
+                continue
+            spec = Dependency(
+                name=spec.name,
+                version=update,
+                extras=spec.extras,
+            )
+            updated += "    " + spec.to_pip_string() + "\n"
+        except ValueError:
+            updated += line + "\n"
+    TOX_INI.write_text(updated[:-1], encoding="utf-8")
+
+
+def bump_packages(dependencies: t.Dict[str, str]) -> None:
+    """Bump packages."""
+    _logger.info("Updating packages")
+    manager = PackageManagerV1.from_dir(Path(PACKAGES))
+    for package_id in manager.dev_packages:
+        path = (
+            manager.package_path_from_package_id(
+                package_id=package_id,
+            )
+            / PACKAGE_TYPE_TO_CONFIG_FILE[package_id.package_type.value]
+        )
+        with path.open("r", encoding="utf-8") as stream:
+            config, *extra = yaml_load_all(stream=stream)
+
+        for name in config.get("dependencies", {}):
+            update = dependencies.get(name)
+            if update is None:
+                continue
+            config["dependencies"][name]["version"] = update
+
+        with path.open("w", encoding="utf-8") as stream:
+            yaml_dump_all([config, *extra], stream=stream)
+
+
+@click.command(name="bump")
+@click.option(
+    "-d",
+    "--dependency",
+    "extra",
+    type=PyPiDependency(),
+    multiple=True,
+    help="Specify extra dependency.",
+)
+@click.option("--sync", is_flag=True, help="Perform sync.")
+@click.option(
+    "--no-cache",
+    is_flag=True,
+    default=False,
+    help="Avoid using cache to bump.",
+)
+def main(extra: t.List[Dependency], sync: bool, no_cache: bool) -> None:
+    """Run the bump script."""
+    if not no_cache:
+        load_git_cache()
+
+    dependencies = {}
+    dependencies.update(get_dependencies())
+    dependencies.update({dep.name: dep.version for dep in extra or []})
+
+    bump_pipfile_or_pyproject(PIPFILE, dependencies=dependencies)
+    bump_pipfile_or_pyproject(PYPROJECT_TOML, dependencies=dependencies)
+    bump_tox(dependencies=dependencies)
+    bump_packages(dependencies=dependencies)
+    dump_git_cache()
+
+    if sync:
+        pm = PackageManagerV1.from_dir(
+            Path.cwd() / PACKAGES, config_loader=load_configuration
+        )
+        pm.sync(
+            sources=[
+                f"{OPEN_AEA_REPO}:{_version_cache[OPEN_AEA_REPO]}",
+                f"{OPEN_AUTONOMY_REPO}:{_version_cache[OPEN_AUTONOMY_REPO]}",
+            ]
+        )
```

Comment:
What if we need more sources? For example `contribution` needs IEKit as a source when syncing cc @dvilelaf.

Reply:
Yes, it would be nice if we can specify those